libnavigation-base/src/main/java/com/mapbox/navigation/base/model/route/RouteProgressStateMap.kt

```diff
@@ -0,0 +1,14 @@
+package com.mapbox.navigation.base.model.route
+
+import java.util.HashMap
+
+class RouteProgressStateMap : HashMap<RouteState, RouteProgressState?>() {
```

Comment:
This class is unused in new modules. So I think we need to remove it until we will not need it to use, and only them add it to new modules, because now it is a bit confusing.
Moreover, do we have an option to avoid using this class at all, cause now it looks useless a bit and I think under the hood we can avoid use this map. (Or create Mapper class at least... I think it is better) WDYT?